File: 36.py
from typing import List


def isValidSudoku(board: List[List[str]]) -> bool:
        for i in range(0,len(board)):
            row=board[i]
            element_count={}
            for j in row:
                if j !='.':
                    if(j not in element_count.keys()):
                        element_count[j]=1
                    else:
                        element_count[j]+=1
            print(row)
            print(element_count)
            for k in element_count.values():
                if(k>1):
                    print(k)
                    return False
        for i in range(0,len(board)):
            column_index=i
            element_count={}
            for j in range(0,len(board)):
                if(board[j][column_index]!="."):
                    if(board[j][column_index] not in element_count.keys() ):
                        element_count[board[j][column_index]]=1
                    else:
                        element_count[board[j][column_index]]+=1
            for k in element_count.values():
                if(k>1):
                    return False
        return True

print("Sudoko Validation")
board = [["5","3",".",".","7",".",".",".","."]
,["6",".",".","1","9","5",".",".","."]
,[".","9","8",".",".",".",".","6","."]
,["8",".",".",".","6",".",".",".","3"]
,["4",".",".","8",".","3",".",".","1"]
,["7",".",".",".","2",".",".",".","6"]
,[".","6",".",".",".",".","2","8","."]
,[".",".",".","4","1","9",".",".","5"]
,[".",".",".",".","8",".",".","7","9"]]
print(isValidSudoku(board))

File: 36.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        for (int r = 0; r < 9; r++) {
            if (!isValidRow(board, r)) {
                return false;
            }
        }
        for (int c = 0; c < 9; c++) {
            if (!isValidColumn(board, c)) {
                return false;
            }
        }
        for (int r = 0; r < 9; r += 3) {
            for (int c = 0; c < 9; c += 3) {
                if (!isValidSquare(board, r, c)) {
                    return false;
                }
            }
        }
        return true;
    }

    public boolean isValidRow(char[][] board, int r) {
        Map<Character, Integer> row = new HashMap<>();
        for (int c = 0; c < 9; c++) {
            if (board[r][c] == '.') {
                continue;
            }
            if (row.containsKey(board[r][c])) {
                return false;
            }
            row.put(board[r][c], 1);
        }
        return true;
    }

    public boolean isValidColumn(char[][] board, int c) {
        Map<Character, Integer> column = new HashMap<>();
        for (int r = 0; r < 9; r++) {
            if (board[r][c] == '.') {
                continue;
            }
            if (column.containsKey(board[r][c])) {
                return false;
            }
            column.put(board[r][c], 1);
        }
        return true;
    }

    public boolean isValidSquare(char[][] board, int r, int c) {
        Map<Character, Integer> matrix = new HashMap<>();
        for (int i = r; i < r + 3; i++) {
            for (int j = c; j < c + 3; j++) {
                if (board[i][j] == '.') {
                    continue;
                }
                if (matrix.containsKey(board[i][j])) {
                    return false;
                }
                matrix.put(board[i][j], 1);
            }
        }
        return true;
    }
}

// class Solution {
// public boolean isValidSudoku(char[][] board) {
// Set<String> seen = new HashSet<>();
// for (int r = 0; r < 9; r++) {
// for (int c = 0; c < 9; c++) {
// char num = board[r][c];
// if (num == '.')
// continue;

// // Create unique strings for row, column, and box
// String rowKey = num + " in row " + r;
// String colKey = num + " in col " + c;
// String boxKey = num + " in box " + (r / 3) + "-" + (c / 3);

// // Check if any of these keys already exist in the set
// if (!seen.add(rowKey) || !seen.add(colKey) || !seen.add(boxKey)) {
// return false;
// }
// }
// }
// return true;
// }

// }
